dbos/_context.py

```diff
@@ -388,6 +445,10 @@ def __exit__(
         ctx = assert_current_dbos_context()
         assert ctx.is_within_workflow()
         ctx.end_workflow(exc_value, self.is_temp_workflow)
+        # Restore the saved workflow timeout
+        ctx.workflow_timeout_ms = self.saved_workflow_timeout
```

Comment:
Why do we need to store the saved workflow timeout in two places?

Reply:
I wonder if we should implement similar logic in `EnterDBOSChildWorkflow` as well?

Reply:
Where's the other place we save it? No need to do it in `EnterDBOSChildWorkflow ` as that's always a fresh context.